src/vs/workbench/services/runtimeSession/common/runtimeSession.ts

```diff
@@ -1666,6 +1671,30 @@ export class RuntimeSessionService extends Disposable implements IRuntimeSession
 		return id;
 	}
 
+	private async scheduleUpdateActiveLanguages(delay = 60 * 60 * 1000): Promise<void> {
+		return timeout(delay)
+			.then(() => {
+				this.updateActiveLanguages();
+			})
+			.then(() => {
+				return this.scheduleUpdateActiveLanguages();
+			});
+	}
+
+	public updateActiveLanguages(): void {
+		const languages: string[] = [];
+		this._activeSessionsBySessionId.forEach(activeSession => {
```

Comment:
It seems like this will generate an array with multiple copies of the language ID if there are multiple sessions running for the language. Should we skip the session if the language is already marked active?

Reply:
Good point. I've fixed this.